src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/Fx2Nl/Fx2NlLanguageServerOperationHandler.cs

```diff
@@ -57,14 +57,27 @@ private Task PreHandleFx2NlAsync(LanguageServerOperationContext operationContext
         }
 
         /// <summary>
-        /// Gets the Fx2Nl hints.
+        /// Gets the Fx2Nl hints including Usage hints and optional range.
         /// </summary>
         /// <param name="scope"> PowerFx Scope.</param>
         /// <param name="operationContext">Language Server Operation Context.</param>
         /// <returns>Fx2Nl hints.</returns>
         private Fx2NLParameters GetFx2NlHints(IPowerFxScope scope,  LanguageServerOperationContext operationContext)
         {
-            return scope is IPowerFxScopeFx2NL fx2NLScope ? fx2NLScope.GetFx2NLParameters() : new Fx2NLParameters();
+            if (scope is IPowerFxScopeFx2NL fx2NLScope)
+            {
+                var parameters = fx2NLScope.GetFx2NLParameters();

```

Comment:
you do not need to create the requirement that range should only be put in the fx2nl parameters if scope is of this type or that.

this can be simplified to

var parameters = scope is IPowerFxScopeFx2NL fx2NLScope ? fx2NlScope.GetFx2nlParameters() :  new Fx2NlParameters()

parameters.Range ??= _fx2nlRequestParams.Range

this covers an edge case you are missing here. not all hosts are expected to implement IPowerFxScopeFx2Nl but they still might want range/partial fx2nl and your logic doesn't account for that

Reply:
Ah gotcha, did not know about that edge case and thanks for the explanation. Makes sense. Let me address it soon.